File: openapi-test-kotlin-server-quarkus-reactive-client-mp-reactive/testapi/model/Cat.kt
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotNull
import java.time.OffsetDateTime
import java.util.Optional

data class Cat(
  @JsonProperty("meowVolume") val meowVolume: Optional<Integer>,
  @JsonProperty("id") @NotNull val id: String,
  @JsonProperty("createdAt") @NotNull val createdAt: OffsetDateTime,
  /** Whether the cat is an indoor cat */
  @JsonProperty("indoor") @NotNull val indoor: Boolean,
  @JsonProperty("name") @NotNull val name: String,
  @JsonProperty("updatedAt") val updatedAt: Optional<OffsetDateTime>
) : Animal {
  override fun animal_type(): String = "cat"
}

File: openapi-test-scala-server-spring-client-jdk/testapi/model/Cat.scala
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.OffsetDateTime
import java.util.Optional

case class Cat(
    @JsonProperty("meowVolume") meowVolume: Optional[Integer],
    @JsonProperty("id") id: PetId,
    @JsonProperty("createdAt") createdAt: OffsetDateTime,
    /** Whether the cat is an indoor cat */
    @JsonProperty("indoor") indoor: java.lang.Boolean,
    @JsonProperty("name") name: String,
    @JsonProperty("updatedAt") updatedAt: Optional[OffsetDateTime]
) extends Animal {
  @JsonProperty("animal_type")
  override lazy val animal_type: String = "cat"
}
